controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    diff_drive_controller:
      type: diff_drive_controller/DiffDriveController

    head_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    rarm_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    larm_controller:
      type: joint_trajectory_controller/JointTrajectoryController

diff_drive_controller:
  ros__parameters:
    left_wheel_names: ["front_left_wheel_joint", "rear_left_wheel_joint"]
    right_wheel_names: ["front_right_wheel_joint", "rear_right_wheel_joint"]

    wheel_separation: 0.101
    wheel_radius: 0.030083

    wheel_separation_multiplier: 1.0
    left_wheel_radius_multiplier: 1.0
    right_wheel_radius_multiplier: 1.0

    publish_rate: 50.0
    odom_frame_id: odom
    base_frame_id: base_link
    pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
    twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
    open_loop: true
    enable_odom_tf: true

    cmd_vel_timeout: 0.5
    #publish_limited_velocity: true
    use_stamped_vel: false
    #velocity_rolling_window_size: 10

    # Velocity and acceleration limits
    # Whenever a min_* is unspecified, default to -max_*
    linear.x.has_velocity_limits: true
    linear.x.has_acceleration_limits: true
    linear.x.has_jerk_limits: false
    linear.x.max_velocity: 1.0
    linear.x.min_velocity: -1.0
    linear.x.max_acceleration: 1.0
    linear.x.max_jerk: 0.0
    linear.x.min_jerk: 0.0

    angular.z.has_velocity_limits: true
    angular.z.has_acceleration_limits: true
    angular.z.has_jerk_limits: false
    angular.z.max_velocity: 1.0
    angular.z.min_velocity: -1.0
    angular.z.max_acceleration: 1.0
    angular.z.min_acceleration: -1.0
    angular.z.max_jerk: 0.0
    angular.z.min_jerk: 0.0

head_controller:
  ros__parameters:
    joints:
      - head_joint0
      - head_joint1
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 50.0
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false
    allow_nonzero_velocity_at_trajectory_end: true
    allow_integration_in_goal_trajectories: false
    constraints:
      stopped_velocity_tolerance: 0.1
      goal_time: 2.0
      head_joint0:
        trajectory: 0.5
        goal: 0.1
      head_joint1:
        trajectory: 0.5
        goal: 0.1

rarm_controller:
  ros__parameters:
    joints:
      - rarm_joint0
      - rarm_joint1
      - rarm_joint2
      - rarm_joint3
      - rarm_joint4
      - rarm_joint5
      - rarm_joint6
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 50.0
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false
    constraints:
      stopped_velocity_tolerance: 0.1
      goal_time: 2.0
      rarm_joint0:
        trajectory: 0.5
        goal: 0.1
      rarm_joint1:
        trajectory: 0.5
        goal: 0.1
      rarm_joint2:
        trajectory: 0.5
        goal: 0.1
      rarm_joint3:
        trajectory: 0.5
        goal: 0.1
      rarm_joint4:
        trajectory: 0.5
        goal: 0.1
      rarm_joint5:
        trajectory: 0.5
        goal: 0.1
      rarm_joint6:
        trajectory: 0.5
        goal: 0.1

larm_controller:
  ros__parameters:
    joints:
      - larm_joint0
      - larm_joint1
      - larm_joint2
      - larm_joint3
      - larm_joint4
      - larm_joint5
      - larm_joint6
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 50.0
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false
    constraints:
      stopped_velocity_tolerance: 0.1
      goal_time: 2.0
      larm_joint0:
        trajectory: 0.5
        goal: 0.1
      larm_joint1:
        trajectory: 0.5
        goal: 0.1
      larm_joint2:
        trajectory: 0.5
        goal: 0.1
      larm_joint3:
        trajectory: 0.5
        goal: 0.1
      larm_joint4:
        trajectory: 0.5
        goal: 0.1
      larm_joint5:
        trajectory: 0.5
        goal: 0.1
      larm_joint6:
        trajectory: 0.5
        goal: 0.1
